bench-vortex/src/tpch_benchmark.rs

```diff
@@ -0,0 +1,290 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: Copyright the Vortex contributors
+
+//! TPCH benchmark implementation
+
+use std::path::{Path, PathBuf};
+use std::{env, fs};
+
+use anyhow::{Result, anyhow};
+use log::{info, warn};
+use similar::{ChangeTag, TextDiff};
+use url::Url;
+use vortex::error::VortexExpect;
+
+use crate::benchmark_trait::Benchmark;
+use crate::engines::{EngineCtx, ddb};
+use crate::tpch::duckdb::{DuckdbTpcOptions, TpcDataset, generate_tpc};
+use crate::tpch::{EXPECTED_ROW_COUNTS_SF1, EXPECTED_ROW_COUNTS_SF10, tpch_queries};
+use crate::{BenchmarkDataset, Format, IdempotentPath, Target};
+
+/// TPCH benchmark implementation
+pub struct TpcHBenchmark {
+    pub scale_factor: u32,
+    pub use_remote_data_dir: Option<String>,
+}
+
+impl TpcHBenchmark {
+    pub fn new(scale_factor: u32, use_remote_data_dir: Option<String>) -> Self {
+        Self {
+            scale_factor,
+            use_remote_data_dir,
+        }
+    }
+}
+
+impl Benchmark for TpcHBenchmark {
+    fn queries(&self) -> Result<Vec<(usize, String)>> {
+        Ok(tpch_queries().collect())
+    }
+
+    fn generate_data(&self, _data_url: &Url, target: &Target) -> Result<()> {
+        match &self.use_remote_data_dir {
+            None => {
+                // Generate data for the specific target format (idempotent)
+                let format = if target.format() == Format::Arrow {
+                    Format::Csv
+                } else {
+                    target.format()
+                };
+
+                let opts = DuckdbTpcOptions::new("tpch".to_data_path(), TpcDataset::TpcH, format)
```

Comment:
The strings passed to `DuckdbTpcOptions` should probably be enums which inside of `new` construct the full data path.

Reply:
This is a file path, so I am not sure an enum makes sense